src/compas_cgal/polylines.py

```diff
@@ -0,0 +1,98 @@
+"""Polyline utilities using CGAL."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from numpy.typing import NDArray
+
+from compas_cgal import _types_std  # noqa: F401  # Load vector type bindings
+from compas_cgal._polylines import closest_points_on_polyline as _closest_points
+from compas_cgal._polylines import simplify_polylines as _simplify
+
+if TYPE_CHECKING:
+    PointsList = list[list[float]] | NDArray
```

Comment:
just fyi, if you intend to use this in Rhino CPython, the `|` will throw an error. it was only introduced in python 3.10